Services/CO.CDP.Forms.WebApi/UseCase/UpdateFormSectionAnswersUseCase.cs

```diff
@@ -38,18 +38,41 @@ public async Task<bool> Execute((Guid formId, Guid sectionId, Guid answerSetId,
         }
         else
         {
-            existingAnswerSet = new Persistence.FormAnswerSet
+            if (section.Form == null)
             {
-                Guid = answerSetId,
+                throw new Exception($"Form not found.");
+            }
+            if (section.Form.Guid != formId)
+            {
+                throw new Exception($"Form with Guid {formId} not found.");
+            }
+
+
+            var sharedConsent = new Persistence.SharedConsent
```

Comment:
This doesn't seems right, we should not be creating SharedConsent every time a new AnswerSet is created.
It should only be created  when no draft SharedConset exists

Reply:
added an explicit check as discussed in the thread